### ISSUE TITLE
Errors connecting to Mosquitto mqtt broker when enforcing TLS1.3

### ISSUE THREAD

**Issue by AviEpstein** [Mbed-TLS/mbedtls#6489]
### Summary
when trying to connect to Mosquitto mqtt broker and enforcing TLS1.3 im getting the following error:

-------------------------------------------------------------------------------------------------------------------
.
.
.
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_tls13_client.c:1723: 20006e58: tls13 client state: MBEDTLS_SSL_ENCRYPTED_EXTENSIONS(19)
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_tls13_client.c:1462: 20006e58: => parse encrypted extensions
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:3546: 20006e58: => read record
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1573: 20006e58: => fetch input
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1730: 20006e58: in_left: 0, nb_want: 5
[SL-MBEDTLS::TRACE] net_recv_timeout:: fd=1, len=5 (5) 
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1755: 20006e58: in_left: 0, nb_want: 5
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1756: 20006e58: ssl->f_recv(_timeout)() returned 5 (-0xfffffffb)
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1776: 20006e58: <= fetch input
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:3284: 20006e58: input record: msgtype = 20, version = [3:3], msglen = 1
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1573: 20006e58: => fetch input
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1730: 20006e58: in_left: 5, nb_want: 6
[SL-MBEDTLS::TRACE] net_recv_timeout:: fd=1, len=1 (1) 
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1755: 20006e58: in_left: 5, nb_want: 6
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1756: 20006e58: ssl->f_recv(_timeout)() returned 1 (-0xffffffff)
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1776: 20006e58: <= fetch input
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:4463: 20006e58: Ignore ChangeCipherSpec in TLS 1.3 compatibility mode
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1573: 20006e58: => fetch input
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1730: 20006e58: in_left: 0, nb_want: 5
[SL-MBEDTLS::TRACE] net_recv_timeout:: fd=1, len=5 (5) 
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1755: 20006e58: in_left: 0, nb_want: 5
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1756: 20006e58: ssl->f_recv(_timeout)() returned 5 (-0xfffffffb)
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1776: 20006e58: <= fetch input
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:3284: 20006e58: input record: msgtype = 23, version = [3:3], msglen = 27
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1573: 20006e58: => fetch input
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1730: 20006e58: in_left: 5, nb_want: 32
[SL-MBEDTLS::TRACE] net_recv_timeout:: fd=1, len=27 (27) 
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1755: 20006e58: in_left: 5, nb_want: 32
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1756: 20006e58: ssl->f_recv(_timeout)() returned 27 (-0xffffffe5)
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1776: 20006e58: <= fetch input
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1014: 20006e58: => decrypt buf
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1539: 20006e58: <= decrypt buf
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:2664: 20006e58: handshake message: msglen = 10, type = 8, hslen = 10
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:3620: 20006e58: <= read record
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_tls13_client.c:1502: 20006e58: dumping 'encrypted extensions' (4 bytes)
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_tls13_client.c:1502: 20006e58: 0000:  00 00 00 00                                      ....
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_tls13_client.c:1537: 20006e58: unsupported extension found: 0 
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_tls13_client.c:1479: 20006e58: <= parse encrypted extensions
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:4563: 20006e58: => send alert message
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:4564: 20006e58: send alert level=2 message=110
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:2369: 20006e58: => write record
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:2461: 20006e58: output record: msgtype = 21, version = [3:3], msglen = 2
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1789: 20006e58: => flush output
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1809: 20006e58: message length: 7, out_left: 7
[SL-MBEDTLS::TRACE] net_send:: fd=1, len=7 (7) 
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1814: 20006e58: ssl->f_send() returned 7 (-0xfffffff9)
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:1842: 20006e58: <= flush output
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:2514: 20006e58: <= write record
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_msg.c:4576: 20006e58: <= send alert message
[SL-MBEDTLS::TRACE] [?] ../../../../library/ssl_tls.c:5323: 20006e58: <= handshak
[SL-MBEDTLS::ERROR]    Failed ! mbedtls_ssl_handshake returned -0x7500

----------------------------------------------------------------------------------------------------------------------

I tried to look for a fix in your form and i found this:
- github.com/hannestschofenig/mbedtls/blob/tls13-prototype/library/ssl_tls13_client.c
- github.com/Mbed-TLS/mbedtls/issues/4423

and added the following code to ssl_tls1.3_client: 

#if defined(MBEDTLS_SSL_SERVER_NAME_INDICATION)
            case MBEDTLS_TLS_EXT_SERVERNAME:
                MBEDTLS_SSL_DEBUG_MSG( 3, ( "found server_name extension" ) );

                /* The server_name extension should be an empty extension */

                break;
#endif /* MBEDTLS_SSL_SERVER_NAME_INDICATION */

I don't know if this is an allowed fix. but when using the fix I'm able to successfully verify the server and i get a success return from mbedtls_ssl_handshake(*) and mbedtls_ssl_get_verify_result()

however a new error appears:

**--------------------------------------------------------------------------------------------------------------
.
.
.
[SL-MBEDTLS::INFO]     Success ! SSL/TLS handshake completed !
[SL-MBEDTLS::INFO]  Verifying peer X.509 certificate...
[SL-MBEDTLS::INFO]     Success !
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:5689: 20006e58: => write
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2633: 20006e58: => write record
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:0559: 20006e58: => encrypt buf
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:0822: 20006e58: before encrypt: msglen = 64, including 0 bytes of padding
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1118: 20006e58: <= encrypt buf
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2724: 20006e58: output record: msgtype = 23, version = [3:3], msglen = 80
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2016: 20006e58: => flush output
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2036: 20006e58: message length: 85, out_left: 85
[SL-MBEDTLS::TRACE] net_send:: fd=1, len=85 (85) 
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2041: 20006e58: ssl->f_send() returned 85 (-0xffffffab)
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2069: 20006e58: <= flush output
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2777: 20006e58: <= write record
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:5713: 20006e58: <= write
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:5407: 20006e58: => read
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:0068: 20006e58: set_timer to 5000 ms
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:3842: 20006e58: => read record
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1800: 20006e58: => fetch input
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1957: 20006e58: in_left: 0, nb_want: 5
[SL-MBEDTLS::TRACE] net_recv_timeout:: fd=1, len=5 (5) 
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1982: 20006e58: in_left: 0, nb_want: 5
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1983: 20006e58: ssl->f_recv(_timeout)() returned 5 (-0xfffffffb)
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2003: 20006e58: <= fetch input
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:3575: 20006e58: input record: msgtype = 23, version = [0x303], msglen = 74
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1800: 20006e58: => fetch input
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1957: 20006e58: in_left: 5, nb_want: 79
[SL-MBEDTLS::TRACE] net_recv_timeout:: fd=1, len=74 (74) 
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1982: 20006e58: in_left: 5, nb_want: 79
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1983: 20006e58: ssl->f_recv(_timeout)() returned 74 (-0xffffffb6)
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2003: 20006e58: <= fetch input
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1144: 20006e58: => decrypt buf
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1766: 20006e58: <= decrypt buf
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2930: 20006e58: handshake message: msglen = 57, type = 4, hslen = 57
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:3916: 20006e58: <= read record
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:5316: 20006e58: handshake received (not HelloRequest)
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:5498: 20006e58: ssl_handle_hs_message_post_handshake() returned -30464 (-0x7700)
[SL-MBEDTLS::ERROR]  mbedtls_ssl_read returned -30464
-----------------------------------------------------------------------------------------------------------------------------------------**

### What could be the reason for this and how can i solve this issue?

it seems the server is sending un unexpected handshake message after the handshake is finished.

**note:** 
1) this issue only comes up when enforcing tls1.3, if I enforce tls1.2 the application works successfully.
2) I was able to implement the mbedtls interface on http_client app forcing TLS1.3 and it worked seamlessly (with the same configs and system, connecting to a google server)  

### System information
implementing mqtt client on a TI simplelink device (cc3235sf) with the mbedtls 3.2.1 as an interface for the TLS layer.
 enforcing tls1.3 only!

Mbed TLS version (number or commit id): 3.2.1
Operating system and version: RTOS
Configuration (if not default, please attach `mbedtls_config.h`): 

https://tidrive.ext.ti.com/u/-tUvqFTVL6zo1S-m/9f89ac21-562a-4e53-95ba-11c936eb233d?l

Compiler and options (if you used a pre-built binary, please indicate how you obtained it): 
Additional environment information: 

### Expected behavior
successful connection and communication to a mosquitto sever

### Actual behavior
1) failure on server encrypted extensions with extension type "0" (sever name)
2) after implementing a patch failure on post handshake message

### Steps to reproduce
connect to mosquitto broker using a mbedtls as a mqttclient

**Comment by ronald-cron-arm** [comment]
Hi. The first issue you report is fixed in development. Regarding 2, my guess is that the server is sending a new session ticket message and this is not supported in 3.2.1 but supported in development. Thus with development you may be able to connect successfully.

**Comment by daverodgman** [comment]
Closing - please re-open if using development does not resolve the issues.

**Comment by daverodgman** [comment]
(Please note that we're planning a release in December, so assuming using development does resolve the issues, you will be able to pick up the fixes in the upcoming 3.3 release)

**Comment by AviEpstein** [comment]
Thank you for the replay!

I have updated to development version.
As Ronald mentioned it seems the server is sending a new session ticket. 

however in the ssl_tls_1.3_client.c flow the function:

```
int mbedtls_ssl_tls13_handshake_client_step( mbedtls_ssl_context *ssl )
{
.
.
.
#if defined(MBEDTLS_SSL_SESSION_TICKETS)
        case MBEDTLS_SSL_NEW_SESSION_TICKET:
            ret = ssl_tls13_process_new_session_ticket( ssl );
            if( ret != 0 )
                break;
            ret = MBEDTLS_ERR_SSL_RECEIVED_NEW_SESSION_TICKET;
            break;
#endif /* MBEDTLS_SSL_SESSION_TICKETS */
```

**ssl_tls13_process_new_session_ticket(..) returns 0 and hence ret is assigned to MBEDTLS_ERR_SSL_RECEIVED_NEW_SESSION_TICKET which is returned to the app via mbedtls_ssl_read() which called it**

**Is this returned error intentional (like MBEDTLS_ERR_SSL_WANT_READ or MBEDTLS_ERR_SSL_WANT_WRITE) and should be dealt with on the app layer or should it be dealt with internally?**

**If the first case is true I sow that in sslclient2.c MBEDTLS_ERR_SSL_RECEIVED_NEW_SESSION_TICKET is one of the cases.
I added:**

```
#if defined(MBEDTLS_SSL_PROTO_TLS1_3)

#if defined(MBEDTLS_SSL_SESSION_TICKETS)

            case MBEDTLS_ERR_SSL_RECEIVED_NEW_SESSION_TICKET:
                /* We were waiting for application data but got
                 * a NewSessionTicket instead. */
                mbedtls_info( " got new session ticket ( %d ).\n",
                                pWifiSock->ticket_id++ );

                mbedtls_info("  . Saving session for reuse..." );
                if( RECO_MODE == 1 )
                {
                    if( ( ret = ssl_save_session_serialize( &pTlsSock->ssl,
                                                            &pTlsSock->session_data, &pTlsSock->session_data_len ) ) != 0 )
                    {
                        mbedtls_error( " failed\n  ! ssl_save_session_serialize returned -0x%04x\n\n",
                                        (unsigned int) -ret );
                    }
                }
                else
                {
                    if( ( ret = mbedtls_ssl_get_session( &pTlsSock->ssl, &pTlsSock->saved_session ) ) != 0 )
                    {
                        mbedtls_error( " failed\n  ! mbedtls_ssl_get_session returned -0x%x\n\n",
                                        (unsigned int) -ret );
                    }
                }

                mbedtls_trace( " ok after read\n" );

                if( RECO_MODE == 1 )
                {
                    mbedtls_info( "    [ Saved %u bytes of session data]\n",
                                    (unsigned) pTlsSock->session_data_len );
                }
              break;

#endif /* MBEDTLS_SSL_SESSION_TICKETS */

#endif /* MBEDTLS_SSL_PROTO_TLS1_3 */
```

**After the looping call to mbedtls_ssl_read() but still was not able to get any different outcome.
do you know what could be causing this and how could I go on to fix it?**

logs from running the app:

--------------------------------------------------------------------------------------------------------------------------


[SL-MBEDTLS::INFO]     Success ! SSL/TLS handshake completed !
[SL-MBEDTLS::INFO]  Verifying peer X.509 certificate...
[SL-MBEDTLS::INFO]     Success !
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:5750: 20006e58: => write
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2629: 20006e58: => write record
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:0553: 20006e58: => encrypt buf
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:0816: 20006e58: before encrypt: msglen = 64, including 0 bytes of padding
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1112: 20006e58: <= encrypt buf
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2720: 20006e58: output record: msgtype = 23, version = [3:3], msglen = 80
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2012: 20006e58: => flush output
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2032: 20006e58: message length: 85, out_left: 85
[SL-MBEDTLS::TRACE] net_send:: fd=1, len=85 (85) 
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2037: 20006e58: ssl->f_send() returned 85 (-0xffffffab)
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2065: 20006e58: <= flush output
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2773: 20006e58: <= write record
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:5774: 20006e58: <= write
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:5468: 20006e58: => read
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:0062: 20006e58: set_timer to 20000 ms
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:3838: 20006e58: => read record
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1796: 20006e58: => fetch input
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1953: 20006e58: in_left: 0, nb_want: 5
[SL-MBEDTLS::TRACE] net_recv_timeout:: fd=1, len=5 (5) 
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1978: 20006e58: in_left: 0, nb_want: 5
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1979: 20006e58: ssl->f_recv(_timeout)() returned 5 (-0xfffffffb)
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1999: 20006e58: <= fetch input
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:3571: 20006e58: input record: msgtype = 23, version = [0x303], msglen = 74
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1796: 20006e58: => fetch input
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1953: 20006e58: in_left: 5, nb_want: 79
[SL-MBEDTLS::TRACE] net_recv_timeout:: fd=1, len=74 (74) 
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1978: 20006e58: in_left: 5, nb_want: 79
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1979: 20006e58: ssl->f_recv(_timeout)() returned 74 (-0xffffffb6)
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1999: 20006e58: <= fetch input
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1140: 20006e58: => decrypt buf
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1762: 20006e58: <= decrypt buf
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2926: 20006e58: handshake message: msglen = 57, type = 4, hslen = 57
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:3912: 20006e58: <= read record
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:5314: 20006e58: received post-handshake message
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:5302: 20006e58: NewSessionTicket received
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:5468: 20006e58: => read
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls.c:3438: 20006e58: => handshake
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2012: 20006e58: => flush output
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2024: 20006e58: <= flush output
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls.c:3359: 20006e58: client state: MBEDTLS_SSL_SERVER_NEW_SESSION_TICKET
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls13_client.c:2682: 20006e58: => parse new session ticket
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:3838: 20006e58: => read record
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:3908: 20006e58: reuse previously read message
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:3912: 20006e58: <= read record
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls13_client.c:2540: 20006e58: ticket_lifetime: 7200
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls13_client.c:2545: 20006e58: ticket_age_add: 3015721758
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls13_client.c:2552: 20006e58: dumping 'ticket_nonce:' (8 bytes)
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls13_client.c:2552: 20006e58: 0000:  00 00 00 00 00 00 00 00                          ........
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls13_client.c:2560: 20006e58: dumping 'received ticket' (32 bytes)
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls13_client.c:2560: 20006e58: 0000:  b8 74 1a 90 e4 2e fd 0d 6f 60 07 e6 d2 63 eb f1  .t......o`...c..
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls13_client.c:2560: 20006e58: 0010:  61 9e e8 2e 34 30 d8 09 54 17 5f 9e 37 d4 63 d1  a...40..T._.7.c.
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls13_client.c:2585: 20006e58: dumping 'ticket extension' (0 bytes)
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls13_client.c:2636: 20006e58: dumping 'resumption_master_secret' (48 bytes)
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls13_client.c:2636: 20006e58: 0000:  89 fc c9 d1 a8 78 b8 a9 b7 ec 72 36 be 8f 75 b7  .....x....r6..u.
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls13_client.c:2636: 20006e58: 0010:  e7 c7 c7 54 3d c2 47 6a d0 e1 89 5c aa 1d 00 48  ...T=.Gj...\...H
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls13_client.c:2636: 20006e58: 0020:  64 78 47 cb 52 52 a8 c7 8e 9b 2e 4e 21 45 7e 50  dxG.RR.....N!E~P
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls13_client.c:2665: 20006e58: dumping 'Ticket-resumed PSK' (48 bytes)
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls13_client.c:2665: 20006e58: 0000:  f6 fe 48 91 4c 94 a4 bb 4c 85 b4 39 bd df 0d d1  ..H.L...L..9....
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls13_client.c:2665: 20006e58: 0010:  79 17 22 d3 e3 29 4f 75 84 11 07 5b 51 b6 5d a6  y."..)Ou...[Q.].
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls13_client.c:2665: 20006e58: 0020:  ab 56 66 9d 3f 34 5a 83 ad f3 17 ca 56 df 04 b7  .Vf.?4Z.....V...
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls13_client.c:2699: 20006e58: <= parse new session ticket
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls.c:3449: 20006e58: <= handshake
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:5513: 20006e58: mbedtls_ssl_handshake() returned -31488 (-0x7b00)
[SL-MBEDTLS::INFO]   got new session ticket ( 0 ).
[SL-MBEDTLS::INFO]    . Saving session for reuse...
[SL-MBEDTLS::TRACE]  ok
[SL-MBEDTLS::INFO]      [ Saved 130 bytes of session data]
[SL-MBEDTLS::ERROR]  mbedtls_ssl_read returned 0
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:5789: 20006e58: => write close notify
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:4867: 20006e58: => send alert message
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:4868: 20006e58: send alert level=1 message=0
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2629: 20006e58: => write record
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:0553: 20006e58: => encrypt buf
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:0816: 20006e58: before encrypt: msglen = 16, including 0 bytes of padding
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:1112: 20006e58: <= encrypt buf
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2720: 20006e58: output record: msgtype = 23, version = [3:3], msglen = 32
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2012: 20006e58: => flush output
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2032: 20006e58: message length: 37, out_left: 37
[SL-MBEDTLS::TRACE] net_send:: fd=1, len=37 (37) 
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2037: 20006e58: ssl->f_send() returned 37 (-0xffffffdb)
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2065: 20006e58: <= flush output
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:2773: 20006e58: <= write record
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:4880: 20006e58: <= send alert message
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_msg.c:5802: 20006e58: <= write close notify
[SL-MBEDTLS::INFO]  SlNetIfWifi_close:: Connection Socket 0 was removed
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls.c:4356: 20006e58: => free
[SL-MBEDTLS::INFO]  [?] ../../../../library/ssl_tls.c:4424: 20006e58: <= free
[SL-MBEDTLS::INFO]  SlNetIfWifi_close:: Listener Socket 0 was removed
[MQTT_APP::INFO]  MQTT_EVENT_SERVER_DISCONNECT


---------------------------------------------------------------------------------------------------------------------


link to config file:
https://tidrive.ext.ti.com/u/7wLTsUEC07T9AtO_/9e7d02dc-eb6b-41b5-baf6-4d1f33ce6e68?l
